docs/language.rst

```diff
@@ -177,6 +177,24 @@ Lists of primitive types are also primitive types: ``string[]``, ``number[]``, `
     # incorrect, can't assign to dict after construction
     # a["otherkey"] = "othervalue"
 
+Strings
```

Comment:
Something that our documentation is clearly lacking, is a good overview about which different types of string that exist, what their syntax looks like and how they behave with respect to variable expansion and backslash escaping. At the moment this information is spread over the `Strings`, `string interpolation` and `string formatting` sections. I would like to have one section that gives the entire overview, replacing the existing sections. 

I think we have the following types of string:

* Regular string
* Regular multi-line string
* Raw string
* f-string

I would also explicitly mention that raw string and f-string cannot be used together. Because this is something that is possible in Python, but not in the Inmanta language.